framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -1143,6 +1284,40 @@ private boolean handledByValueChecker(AnnotatedTypeMirror type) {
         }
     }
 
+    /** Determines whether a method is the {@code String.length()} method. */
+    boolean isStringLengthMethod(ExecutableElement method) {
+        return method == lengthMethod;
```

Comment:
Elements are not interned. See [the javadoc here](https://docs.oracle.com/javase/8/docs/api/javax/lang/model/element/Element.html). You should always compare them with `equals` instead of `==`